sdks/python/apache_beam/transforms/ptransform_test.py

```diff
@@ -471,6 +471,24 @@ def test_group_by_key(self):
     assert_that(result, equal_to([(1, [1, 2, 3]), (2, [1, 2]), (3, [1])]))
     pipeline.run()
 
+  def test_group_by_key_reiteration(self):
+    class MyDoFn(beam.DoFn):
+      def process(self, gbk_result):
+        value_list = gbk_result[1]
+        sum_val = 0
+        # Iterate the GBK result for multiple times.
+        for _ in range(0, 17):
+          sum_val += sum(value_list)
+        return (gbk_result[0], sum_val)
+
+    pipeline = TestPipeline()
+    pcoll = pipeline | 'start' >> beam.Create(
+        [(1, 1), (1, 2), (1, 3), (1, 4)])
+    result = (pcoll | 'Group' >> beam.GroupByKey()
+              | 'Reiteration-Sum' >> beam.ParDo(MyDoFn()))
+    assert_that(result, equal_to([1, 170]))
```

Comment:
`equal_to` takes a list of PCollection elements. This should be `equal_to([(1, 170)])`.

Reply:
Ah, turns out the DoFn above is supposed to return a list in py SDK.

Thanks for catching this.  Fixed.

Reply:
FYI, if your DoFn only has a process method, it's generally easier (and clearer) to use `beam.Map` or `beam.FlatMap` instead.